scripts/pylib/twister/twisterlib/reports.py

```diff
@@ -0,0 +1,499 @@
+                                            timestamp = self.timestamp,
+                                            time="0",
+                                            tests="0",
+                                            failures="0",
+                                            errors="0", skipped="0")
+            eleTSPropetries = ET.SubElement(eleTestsuite, 'properties')
+            # Multiple 'property' can be added to 'properties'
+            # differing by name and value
+            ET.SubElement(eleTSPropetries, 'property', name="version", value=version)
+
+            total = 0
+            fails = passes = errors = skips = 0
+            for ts in suites:
+                handler_time = ts.get('execution_time', 0)
+                runnable = ts.get('runnable', 0)
+                duration += float(handler_time)
+
+                ts_status = ts.get('status')
+                # Do not report filtered testcases
+                if ts_status == 'filtered' and not self.env.options.detailed_skipped_report:
+                    continue
+                if full_report:
+                    for tc in ts.get("testcases", []):
+                        status = tc.get('status')
+                        reason = tc.get('reason', ts.get('reason', 'Unknown'))
+                        log = tc.get("log", ts.get("log"))
+
+                        tc_duration = tc.get('execution_time', handler_time)
+                        name = tc.get("identifier")
+                        classname = ".".join(name.split(".")[:2])
+                        fails, passes, errors, skips = self.xunit_testcase(eleTestsuite,
+                            name, classname, status, ts_status, reason, tc_duration, runnable,
+                            (fails, passes, errors, skips), log, True)
+                else:
+                    reason = ts.get('reason', 'Unknown')
+                    name = ts.get("name")
+                    classname = f"{platform}:{name}"
+                    log = ts.get("log")
+                    fails, passes, errors, skips = self.xunit_testcase(eleTestsuite,
+                        name, classname, ts_status, ts_status, reason, duration, runnable,
+                        (fails, passes, errors, skips), log, False)
+
+            total = (errors + passes + fails + skips)
+
+            eleTestsuite.attrib['time'] = f"{duration}"
+            eleTestsuite.attrib['failures'] = f"{fails}"
+            eleTestsuite.attrib['errors'] = f"{errors}"
+            eleTestsuite.attrib['skipped'] = f"{skips}"
+            eleTestsuite.attrib['tests'] = f"{total}"
+
+        result = ET.tostring(eleTestsuites)
+        with open(filename, 'wb') as report:
+            report.write(result)
+
+    def json_report(self, filename, version="NA"):
+        logger.info(f"Writing JSON report {filename}")
+        report = {}
+        report["environment"] = {"os": os.name,
+                                 "zephyr_version": version,
```

Comment:
self.env.version

Reply:
this is coming as an argument which is set to self.env.version already.